clippy_lints/src/methods/is_digit_ascii_radix.rs

```diff
@@ -0,0 +1,43 @@
+//! Lint for `c.is_digit(10)`
+
+use super::IS_DIGIT_ASCII_RADIX;
+use clippy_utils::{
+    consts::constant_full_int, consts::FullInt, diagnostics::span_lint_and_sugg, source::snippet_with_applicability,
+};
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'_>,
+    self_arg: &'tcx Expr<'_>,
+    radix: &'tcx Expr<'_>,
+) {
+    if !cx.typeck_results().expr_ty_adjusted(self_arg).peel_refs().is_char() {
+        return;
+    }
+
+    if let Some(radix) = constant_full_int(cx, cx.typeck_results(), radix) {
```

Comment:
[`is_ascii_digit`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_digit) and [is_ascii_hexdigit](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_hexdigit) has been stabilized in `1.24.0` while `is_digit(...)` has been stable since 1.0. Clippy has a configuration that is used to specify what the minimum rust version that the suggestion has to work for.

You can pass the current `msrv` version in as an argument and then check the version like this:

```rs
if meets_msrv(msrv, &msrvs::IS_ASCII_DIGIT) ...
```

See: [*Specifying the lint's minimum supported Rust version (MSRV)*](https://github.com/rust-lang/rust-clippy/blob/master/doc/adding_lints.md#specifying-the-lints-minimum-supported-rust-version-msrv)

They also have a minimum rust version for const evaluation, however, since `is_digit` isn't const we don't have to check for that :upside_down_face: